### ISSUE TITLE
Should not depend on python-typing for Python versions 3.5 and above

### ISSUE THREAD

**Issue by chrbayer** [rizsotto/scan-build#104]
Typing in included in the Python standard library and should thus not be an external dependency for these versions, see: https://pypi.org/project/typing

This leads to dependency problems in Arch Linux and maybe other distributions.

**Comment by rizsotto** [comment]
Thanks for the report Christoph. I think it was reported recently as #92 . This is fixed in version 2.0.14.
Please reopen the issue if you have problem with the latest version.

**Comment by jameysharp** [comment]
To build version 2.0.16, I had to edit `setup.py` in a similar way to the change you made to `requirements.txt`. I replaced `install_requires=['typing']` with `install_requires=['typing; python_version < "3.5"']`.

Without that change, building using `setup.py` failed because I didn't have the `typing` package in my virtualenv, but including `typing` in my virtualenv caused running this package to fail with this message:

```
AttributeError: type object 'Callable' has no attribute '_abc_registry'
```

I think the corresponding change in `requirements.txt` was not quite right, either. As I understand it, the `typing` package is only required for versions of Python older than 3.5, but `requirements.txt` says it is also required for 3.5. I think the version constraint should have been `<` rather than `<=`.
